src/pipeline_migration/actions/migrate.py

```diff
@@ -101,6 +103,65 @@ class InvalidRenovateUpgradesData(ValueError):
     """Raise this error if any required data is missing in the given Renovate upgrades"""
 
 
+def only_tags_pinned_by_version_revision(tags_info: Iterable[dict]) -> Generator[dict, Any, None]:
+    regex = re.compile(TASK_TAG_REGEXP)
+    for tag_info in tags_info:
+        if regex.match(tag_info["name"]):
+            yield tag_info
+
+
+def filter_out_bundles_built_from_older_version(
+    tags_info: Iterable[dict], stop_at_digest: str
+) -> Generator[dict, Any, None]:
+    """Filter out bundles built from older versions
+
+    Generally, once a new version is bumped for a task, there should be no new
+    bundle build for the older versions. However, the corner case happened.
+    Refer to STONEBLD-3667.
+
+    Bundles built for older versions can be ignored because it does not make
+    sense to add migrations to an older version as Konflux users should always
+    follow the newer versions.
+
+    :param tags_info: tags information responded by Quay.io listRepoTags endpoint.
+        Each tag mapping must have a tag name pinned by version and revision, for
+        example, ``0.2-<commit hash>``.
```

Comment:
Is this required for the migration tool to work? Now with the decentralization going on, we probably should not assume task bundles will always have the revision tags.

Reply:
Yes.
> Now with the decentralization going on, we probably should not assume task bundles will always have the commit hash tags.

There is a story created for this. STONEBLD-3673

Reply:
Ok, let's fix it later then